### PR TITLE
fix: Option<T> generates Optional<T> fields instead of PlutusData (#573)

### PR DESCRIPTION
**Pull request by matiwinnetou**
**Problem:**
Blueprint fields referencing Option<T> definitions generated untyped PlutusData
instead of type-safe Optional<T>, requiring manual casting and losing compile-time
type safety. This affected both Aiken v1.0.26 (Option$T) and v1.1.21+ (Option<T>)
syntax, causing SundaeSwap V2/V3 blueprints to have opaque PlutusData fields where
typed Optional<T> was expected.

**Root Cause:**
PlutusBlueprintLoader only detected Option$ prefix (line 119), missing Option<
syntax from newer Aiken versions. When Option<T> definitions weren't marked with
dataType=option, they bypassed OptionDataTypeProcessor and fell back to PlutusData.

**Solution:**

1. **PlutusBlueprintLoader.java** - Extended Option detection to both syntaxes:
   - Added definitions resolution loop (lines 59-70) to set dataType=option
   - Extended check from `ref.startsWith("Option$")` to include `"Option<"`
   - Added null check for blueprints without definitions
   - Now routes all Option types through OptionDataTypeProcessor → Optional<T>

2. **BlueprintAnnotationProcessor.java** - Smart generic type handling:
   - Added resolveDefinitionKeyForClassGeneration() to distinguish built-in
     containers (List, Option, Tuple) from domain types (Interval, ValidityRange)
   - Built-in containers return null (skip generation), domain types return base
     type for typed class generation
   - Supports both $ and <> syntax across Aiken v1.0.x and v1.1.x
   - Changed methods to package-private for testing (no reflection needed)

3. **FieldSpecProcessor.java** - Removed buggy workaround:
   - Reverted resolveBuiltInContainerTypeName() added by previous attempt
   - Simplified PlutusData fallback for truly unresolvable types
   - Now relies on PlutusBlueprintLoader fix for proper type routing

**Test Coverage:**

1. **PlutusBlueprintLoaderTest.java** - New OptionTypeResolution nested class:
   - Tests Option$ and Option< syntax both set dataType=option
   - Verifies Some/None anyOf variants preserved after resolution
   - Tests null definitions handling (no NPE)
   - Validates SundaeSwap V2 and V3 blueprint loading

2. **BlueprintAnnotationProcessorTest.java** - Consolidated comprehensive tests:
   - 56 tests organized in @Nested classes (no reflection, uses package-private)
   - IsBuiltInGenericContainerTests (9 built-ins, 7 domain types, 3 edge cases)
   - ResolveDefinitionKeyForClassGenerationTests (30+ tests covering $ and <>)
   - GenericTypeSkipTests (integration tests for compilation success)
   - Deleted old reflection-based test files

3. **FieldSpecProcessorTest.java** - Fixed misleading comments:
   - Clarified resolveClassNameFromRef() tests class generation, not field typing
   - Corrected "PlutusData fallback" comments to explain Option<T> → Optional<T>
   - Added notes distinguishing skip generation from field type resolution

4. **SundaeSwapV2Test.java & SundaeSwapV3Test.java** - PlutusData assertions:
   - Added verifyNoOpaqueTypes() to detect illegitimate PlutusData fields
   - Filters out legitimate uses (extension/data fields per CIP-57)
   - Ensures Option<T> generates Optional<T>, not PlutusData

**Impact:**

Before (❌):
```java
private PlutusData stakeCredential;  // Option<Credential> → untyped
private PlutusData validityRange;    // Interval$Int → untyped
```

After (✅):
```java
private Optional<StakeCredential> stakeCredential;  // Option<Credential> → typed
private Interval<Integer> validityRange;                     // Interval$Int → typed
```

**Real-World Validation:**
- SundaeSwap V2 (Aiken v1.0.26): Interval$Int generates typed Interval class
- SundaeSwap V3 (Aiken v1.1.21): Option<Credential> generates Optional<Credential>
- All tests pass: ./gradlew :annotation-processor:test :plutus:test

**Breaking Changes:** None - existing blueprints gain better type safety

### CODE REVIEW

File: annotation-processor/src/main/java/com/bloxbean/cardano/client/plutus/annotation/processor/blueprint/FieldSpecProcessor.java [hunk 1]
```diff
@@ -105,6 +105,128 @@ protected String resolveTitleFromDefinitionKey(String definitionKey, BlueprintSc
...
+     */
+    private String extractBaseType(String key) {
+        if (key == null || key.isEmpty()) {
+            return key;
+        }
+
+        // Strip generic parameters: "Interval$Int" → "Interval"
+        int dollarIndex = key.indexOf('$');
+        if (dollarIndex > 0) {
+            key = key.substring(0, dollarIndex);
```
Comment (satran004) [inline annotation-processor/src/main/java/com/bloxbean/cardano/client/plutus/annotation/processor/blueprint/FieldSpecProcessor.java]: Can't we return from here?

File: annotation-processor/src/main/java/com/bloxbean/cardano/client/plutus/annotation/processor/blueprint/FieldSpecProcessor.java [hunk 2]
```diff
@@ -105,6 +105,128 @@ protected String resolveTitleFromDefinitionKey(String definitionKey, BlueprintSc
...
+        }
+
+        // Strip angle bracket generics: "Option<T>" → "Option"
+        int angleIndex = key.indexOf('<');
+        if (angleIndex > 0) {
+            key = key.substring(0, angleIndex);
```
Comment (satran004) [inline annotation-processor/src/main/java/com/bloxbean/cardano/client/plutus/annotation/processor/blueprint/FieldSpecProcessor.java]: Same.
